test/test.js

```diff
@@ -577,4 +577,121 @@ describe('finalhandler(req, res)', function () {
+          // Warning: this may or may not work for your local environment
+          var pathRegex = /\/Users(?:\/[^/]+){3}/g;
+
+          var actualTransformedHTMLText = response.text
+            .replaceAll(pathRegex, '<root>')
+            .replaceAll('&nbsp;', '')
+
+          assert.equal(actualTransformedHTMLText, expectedHTMLText)
+          done()
+        })
+        .catch(done)
+    })
+
+    it('should return Error message with 2 level cause trace', function (done) {
+      var err = new Error('foo', { cause: new Error('bar', { cause: new Error('baz') }) })
+
+      var expectedError = `Error: foo
+  at Context.&lt;anonymous&gt; (<root>/finalhandler/test/test.js:638:17)
+  at callFnAsync (<root>/finalhandler/node_modules/mocha/lib/runnable.js:394:21)
+  ... 7 lines matching cause stack trace ...
+  at process.processImmediate (node:internal/timers:483:21) {
+ [cause]: Error: bar
+   at Context.&lt;anonymous&gt; (<root>/finalhandler/test/test.js:638:43)
+   at callFnAsync (<root>/finalhandler/node_modules/mocha/lib/runnable.js:394:21)
+   ... 7 lines matching cause stack trace ...
+   at process.processImmediate (node:internal/timers:483:21) {
+  [cause]: Error: baz
+    at Context.&lt;anonymous&gt; (<root>/finalhandler/test/test.js:638:69)
+    at callFnAsync (<root>/finalhandler/node_modules/mocha/lib/runnable.js:394:21)
+    at Runnable.run (<root>/finalhandler/node_modules/mocha/lib/runnable.js:338:7)
+    at Runner.runTest (<root>/finalhandler/node_modules/mocha/lib/runner.js:666:10)
+    at <root>/finalhandler/node_modules/mocha/lib/runner.js:789:12
+    at next (<root>/finalhandler/node_modules/mocha/lib/runner.js:581:14)
+    at <root>/finalhandler/node_modules/mocha/lib/runner.js:591:7
+    at next (<root>/finalhandler/node_modules/mocha/lib/runner.js:474:14)
+    at Immediate.&lt;anonymous&gt; (<root>/finalhandler/node_modules/mocha/lib/runner.js:559:5)
+    at process.processImmediate (node:internal/timers:483:21)
+ }
+}`;
+
+		// Transform the error text into a formatted error response - finalhandler adds <br> for \n to make it HTML readable
+		var expectedTransformedError = expectedError.replaceAll(/\n/g, '<br>')
+
+    var expectedHTMLText = `<!DOCTYPE html>
+<html lang="en">
+<head>
+<meta charset="utf-8">
+<title>Error</title>
+</head>
+<body>
+<pre>${expectedTransformedError}</pre>
+</body>
+</html>
+`
+
+      request(createServer(err))
+        .get('/')
+        .then((response) => {
+          // Regular expression to match the root part of any UNIX-style file path
+          // The pathRegex is meant to find/replace the `/Users/.../finalhandler` parts of the expectedError
+          // The pathRegex is also intended to work when ran on other computers and from different root paths.
+          // Warning: this may or may not work for your local environment
+          var pathRegex = /\/Users(?:\/[^/]+){3}/g;
```

Comment:
As with the above comment, this is much too brittle. I would prefer we just use the most simple string contains style check here to ensure the cause error is included. If you like though, you can also see above some tests which dont have so much hard coded assumptions in them.

Reply:
Agreed, what is the best way to go about testing this? I would prefer something a little more explicit in terms of checking the exact output vs just doing a string match against parts of the output.

My reasoning against the string match is that is leaves a lot of assumptions and doesn't necessarily validate that we get the full error output that we want in this case.

I actually had another possibly way to "mask" user/env specific paths via the `process.cwd()` and find/replace it in the final output. What do you think?

Reply:
Let me know if this is better? 39909fb1ab61404dd4268e42b7a42aae5e08c146